.github/workflows/rust.yml

```diff
@@ -0,0 +1,59 @@
+on: [push, pull_request]
+
+name: build
+
+jobs:
+  lint:
+    name: Format and Clippy
+    strategy:
+       matrix:
+         os: [ubuntu-latest, macos-latest]
```

Comment:
Should we also test it on `windows-latest`? With #15 merged, it (should) can be compiled and tested on windows now.

Reply:
If we want to test it on `windows-latest`, we also need to implement `register_signal_handler` for windows. 
Currently, we use `nix` for handling signal, it obviously does not support windows platform.

Reply:
Oh! I forgot. The `setitimer` syscall also couldn't be used on windows :(